app/jobs/ingest_job.rb

```diff
@@ -0,0 +1,17 @@
+class IngestJob < Hyrax::ApplicationJob
+  queue_as Hyrax.config.ingest_queue_name
+
+  after_perform do |job|
+    opts = job.arguments.reverse.detect { |x| x.is_a? Hash } || {}
+    wrapper = job.arguments.first
+    ContentNewVersionEventJob.perform_later(wrapper.file_set, wrapper.user) if opts[:notification]
```

Comment:
@atz :speech_balloon: 
> separate jobs: IngestJob and IngestJobWithNotification. The latter extends the former and unilaterally applies the after_perform block. PRO: options fall away. CON: would need a descendant job for each existing job. If a second (or more) such runtime option emerged, it would become a nonsensical matrix of classes, like IngestJobWithBellsWithNotification and IngestJobWithBellsWithWhistlesWithNotification

Yeah, :-1: 

> some refinement, where the after_perform block moves into a base class (or even the main Hyrax::ActiveJob class). I would consider implementing this when a second job adopts the same pattern, not before.

:+1: to the "after, not before" comment. Now's not the right time for that.

> adding to JobIoWrapper: did not pursue this because it seemed distinct from the local filesystem problem the class intends to solve. The same JobIoWrapper (uploaded file) might be expected to be passed to a series of jobs, some of them with notifications, some without. We shouldn't affect the persisted representation of the wrapper just to reflect that.

OK. What about adding an arg to, or changing the second arg in, `IngestJob` to allow consumers to inject a notification class or the like? That way the code could look like:

```ruby
IngestJob.perform_later(wrapper, notification: true, notification_class: ContentNewVersionEvent)
```

Though maybe we'd need to pass that as a string rather than class name, and the complicating factor here is that event jobs don't all receive the same arguments, so we'd need some way to deal with that. Thinking aloud here.